third_party/gif_fix_image_counter.patch

```diff
@@ -0,0 +1,78 @@
+diff --git a/dgif_lib.c b/dgif_lib.c
+index 82fc097..81f2da4 100644
+--- a/dgif_lib.c
++++ b/dgif_lib.c
+@@ -810,7 +810,8 @@ DGifSetupDecompress(GifFileType *GifFile)
+ 
+     /* coverity[check_return] */
+     if (InternalRead(GifFile, &CodeSize, 1) < 1) {    /* Read Code size from file. */
+-	return GIF_ERROR;    /* Failed to read Code size. */
++        GifFile->Error = D_GIF_ERR_READ_FAILED;
++        return GIF_ERROR;    /* Failed to read Code size. */
+     }
+     BitsPerPixel = CodeSize;
+ 
+@@ -1148,18 +1149,18 @@ DGifSlurp(GifFileType *GifFile)
+               /* Allocate memory for the image */
+               if (sp->ImageDesc.Width <= 0 || sp->ImageDesc.Height <= 0 ||
+                       sp->ImageDesc.Width > (INT_MAX / sp->ImageDesc.Height)) {
+-                  return GIF_ERROR;
++                  goto error;
```

Comment:
We don't allow `goto` or labels.  You can either use a boolean to track success, or a lambda to repeat the error check.

Reply:
@cantonios 
Fixed